tests/ipfwd/test_nhop_group.py

```diff
@@ -371,3 +417,108 @@ def test_nhop(request, duthost, tbinfo):
+    result = asic.run_redis_cmd(
+        argv = ["redis-cli", "-n", 6, "HGETALL", "SWITCH_CAPABILITY|switch"]
+    )
+    it = iter(result)
+    switch_capability = dict(zip(it, it))
+
+    result = asic.run_redis_cmd(
+            argv = ["redis-cli", "-n", 0, "HGETALL", "SWITCH_TABLE:switch"]
+    )
+
+    it = iter(result)
+    switch_table = dict(zip(it, it))
+
+    order_ecmp_capability = switch_capability.get("ORDERED_ECMP_CAPABLE")
+    order_ecmp_configured = switch_table.get("ordered_ecmp")
+    pytest_assert(order_ecmp_capability == order_ecmp_configured, "Order Ecmp Feature configured and capability not same")
+
+    if order_ecmp_configured == "false":
+        pytest.skip("Order ECMP is not configured so skipping the test-case")
+
+    # Check Gather facts IP Interface is active one
+    ip_ifaces = asic.get_active_ip_interfaces(tbinfo).keys()
+    pytest_assert(len(ip_ifaces), "No IP interfaces found")
+    pytest_assert(gather_facts['src_router_intf_name'] in ip_ifaces, "Selected IP interfaces is not active")
+
+    # Generate ARP entries
+    arp_count = 8
+    arplist = Arp(duthost, asic, arp_count, gather_facts['src_router_intf_name'])
+    neighbor_mac = [neighbor[1].lower() for neighbor in arplist.ip_mac_list]
+    original_ip_mac_list = arplist.ip_mac_list[:]
+    ip_route = "192.168.100.50"
+    ip_prefix = ip_route + "/31"
+    ip_ttl = 121
+
+    # create nexthop group
+    nhop = IPRoutes(duthost, asic)
+
+    recvd_pkt_result = defaultdict(int)
+
+    rtr_mac= asic.get_router_mac()
+    
+    pkt, exp_pkt = build_pkt(rtr_mac, ip_route, ip_ttl)
+
+    for x in range(2):
+        try:
+            # create neighbor entry randomly
+            random.shuffle(arplist.ip_mac_list)
+            arplist.arps_add()
+            ips = [arplist.ip_mac_list[x].ip for x in range(arp_count)]
+
+            # add IP route with the next hop group created with nexthop added randomly
+            random.shuffle(ips)
+            nhop.add_ip_route(ip_prefix, ips)
+
+            nhop.program_routes()
+            # wait for routes to be synced and programmed
+            time.sleep(5)
+
+            ptfadapter.dataplane.flush()
+            testutils.send(ptfadapter, gather_facts['dst_port_ids'][0], pkt, 10)
+            (recv_port_idx, recv_pkt) = testutils.verify_packet_any_port(test=ptfadapter, pkt=exp_pkt, ports=gather_facts['src_port_ids'])
+            
+            assert recv_pkt
+
+            # Make sure routing is done
+            pytest_assert(Ether(recv_pkt).ttl == (ip_ttl - 1), "Routed Packet TTL not decremented")
+            pytest_assert(Ether(recv_pkt).src == rtr_mac, "Routed Packet Source Mac is not router MAC")
+            pytest_assert(Ether(recv_pkt).dst.lower() in neighbor_mac, "Routed Packet Destination Mac not valid neighbor entry")
+            # Add the receive port index and reviced dest mac (Nexthop identify property) to the dictionary
+            recvd_pkt_result[(recv_port_idx,Ether(recv_pkt).dst)] += 1
```

Comment:
We can remove the port_indx from the key to avoid Lag hash inconsistency.

Reply:
@prsunny done